src/rust/container/lib.rs

```diff
@@ -0,0 +1,384 @@
+                19 => "SIGSTOP",
+                20 => "SIGTSTP",
+                _ => "SIGTERM", // Default fallback
+            };
+
+            let options = StopContainerOptionsBuilder::default().signal(signal_name);
+            docker
+                .stop_container(&container_name, Some(options.build()))
+                .await
+                .map_err(|e| ContainerError::DestroyError(e.to_string()))?;
+            Ok(())
+        })
+    }
+
+    /// Starts a container.
+    /// This method doesn't handle enableInternet attribute since
+    /// it doesn't make sense for the local usage, and will be hard to
+    /// manage while still having port mapping to the host.
+    fn start(
+        &mut self,
+        raw_params: container::StartParams,
+        _: container::StartResults,
+    ) -> Promise<(), capnp::Error> {
+        let container_name = self.container_name.clone();
+        let docker = self.docker.clone();
+        Promise::from_future(async move {
+            let params = raw_params.get()?;
+            let entrypoint = params
+                .get_entrypoint()?
+                .iter()
+                .map(|s| s.unwrap().to_string().unwrap())
+                .collect::<Vec<String>>();
+            let env = params
+                .get_environment_variables()?
+                .iter()
+                .map(|s| s.unwrap().to_string().unwrap())
+                .collect::<Vec<String>>();
+            let mut create_config = CreateExecOptions::default();
+            create_config.cmd = Some(entrypoint);
+            create_config.env = Some(env);
+            let result = docker
+                .create_exec(&container_name, create_config)
+                .await
+                .map_err(|e| ContainerError::ExecError(e.to_string()))?;
+
+            let start_config = StartExecOptions::default();
+            docker
+                .start_exec(&result.id, Some(start_config))
+                .await
+                .map_err(|e| ContainerError::ExecError(e.to_string()))?;
+            // TODO(soon): Do something with the result.id
+            Ok(())
+        })
+    }
+
+    fn status(
+        &mut self,
+        _: container::StatusParams,
+        mut results: container::StatusResults,
+    ) -> Promise<(), capnp::Error> {
+        let container_name = self.container_name.clone();
+        let docker = self.docker.clone();
+        Promise::from_future(async move {
+            let inspect = docker
+                .inspect_container(
+                    &container_name,
+                    None::<bollard::query_parameters::InspectContainerOptions>,
+                )
+                .await
+                .map_err(|e| ContainerError::ConnectionError(e.to_string()))?;
+
+            let mut builder = results.get();
+            if let Some(state) = inspect.state {
+                builder.set_running(state.running.unwrap_or(false));
+            }
+            Ok(())
+        })
+    }
+
+    fn monitor(
```

Comment:
I believe this just blocks until the container exits then surfaces the error if the error code is not 0